FirebaseUI/Database/API/FirebaseArray.h

```diff
@@ -39,43 +49,36 @@ NS_ASSUME_NONNULL_BEGIN
  * The delegate object that array changes are surfaced to, which conforms to the
  * [FirebaseArrayDelegate Protocol](FirebaseArrayDelegate).
  */
-@property(weak, nonatomic) id<FirebaseArrayDelegate> delegate;
+@property(weak, nonatomic, nullable) id<FirebaseArrayDelegate> delegate;
 
 /**
  * The query on a Firebase reference that provides data to populate the instance of FirebaseArray.
  */
-@property(strong, nonatomic) FIRDatabaseQuery *query;
+@property(strong, nonatomic) id<FIRDataObservable> query;
 
 /**
- * The delegate object that array changes are surfaced to.
+ * The number of objects in the FirebaseArray.
  */
-@property(strong, nonatomic) NSMutableArray<FIRDataSnapshot *> * snapshots;
-
-#pragma mark -
-#pragma mark Initializer methods
+@property(nonatomic, readonly) NSUInteger count;
 
 /**
- * Intitalizes FirebaseArray with a standard Firebase reference.
- * @param ref The Firebase reference which provides data to FirebaseArray
- * @return The instance of FirebaseArray
+ * The items currently in the FirebaseArray.
  */
-- (instancetype)initWithRef:(FIRDatabaseReference *)ref;
+@property(nonatomic, readonly, copy) NSArray *items;
```

Comment:
I think it makes sense to keep `copy` since it'll copy on reads. Our style guide doesn't say anything about property annotations, and in general for an open source project that we'll be breaking/changing a lot I think it's healthier to disregard internal style guides and agree on whatever style we'll use with the community.